crates/gpui/src/platform/linux/x11/client.rs

```diff
@@ -1083,6 +1103,24 @@ impl X11Client {
                 let modifiers = modifiers_from_state(event.state);
                 state.modifiers = modifiers;
 
+                // Macros containing modifiers might result in
+                // the modifiers missing from the event.
+                // We therefore update the mask from the global state.
+                let locked_mods_mask = (ModMask::LOCK | ModMask::M2).bits() as u32;
+                let event_state_bits = u32::from(event.state.bits());
+                let depressed_mods = event_state_bits & !locked_mods_mask;
+                let latched_mods = state.xkb.serialize_mods(xkbc::STATE_MODS_LATCHED);
+                let locked_mods = state.xkb.serialize_mods(xkbc::STATE_MODS_LOCKED);
+                let locked_layout = state.xkb.serialize_layout(xkbc::STATE_LAYOUT_LOCKED);
+                state.xkb.update_mask(
+                    depressed_mods,
```

Comment:
This could be simplified

```suggestion
                let depressed_mods = event.state.remove(ModMask::LOCK).remove(ModMask::M2);
                let latched_mods = state.xkb.serialize_mods(xkbc::STATE_MODS_LATCHED);
                let locked_mods = state.xkb.serialize_mods(xkbc::STATE_MODS_LOCKED);
                let locked_layout = state.xkb.serialize_layout(xkbc::STATE_LAYOUT_LOCKED);
                state.xkb.update_mask(
                    depressed_mods.into(),
```

Even without using the `remove` method, converting u16 to u32 until we pass it to `update_mask` later and converting `event.state` to bits isn't necessary, it would've been sufficient to do
```rust
                let locked_mods_mask = ModMask::LOCK | ModMask::M2;
                let depressed_mods = event.state & !locked_mods_mask.bits();
```